### PR TITLE
Fixes #17689: Add an option to exit with an error code if here was an application error

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/17689

### CODE REVIEW

File: share/commands/agent-log [hunk 1]
```diff
@@ -18,6 +18,19 @@
...
+# @man +
+# @man *-E*: exit with an error if there a non compliance
+# @man +
+# @man *Exit codes*:
+# @man +
+# @man *0*: Agent ran normally
+# @man +
+# @man *1*: Agent encountered a critial error and could not run properly
+# @man +
+# @man *2*: Some policy encountered an error
```
Comment (amousset) [inline share/commands/agent-log:31]:
```suggestion
# @man *2*: Some policy encountered and error and *-e* parameter was passed
```

File: share/commands/agent-log [hunk 2]
```diff
@@ -18,6 +18,19 @@
...
+# @man +
+# @man *3*: Some policy encountered a non compliance
```
Comment (amousset) [inline share/commands/agent-log:33]:
```suggestion
# @man *3*: Some policy encountered a non compliance and *-E* parameter was passed
```

**Comment by peckpeck** [comment]
PR updated with a new commit

**Comment by peckpeck** [comment]
OK, squash merging this PR

**Pull request merged**